marklogic-data-hub-central/ui/src/components/entities/create-edit-step/create-edit-step.tsx

```diff
@@ -416,15 +416,15 @@ const CreateEditStep: React.FC<Props>  = (props) => {
             placeholder="Enter name"
             value={stepName}
             onChange={handleChange}
-            disabled={tobeDisabled}
+            disabled={tobeDisabled || props.stepType === StepType.Custom /** custom step names cannot be changes */}
```

Comment:
This condition seems redundant because the name field is always disabled for all steps when editing. I'm thinking it can be removed @jyangML

Reply:
Fixed